pkg/multicluster/register/interfaces.go

```diff
@@ -148,15 +155,15 @@ type ClusterRegistrant interface {
 	) error
 
 	/*
-		Same functionality as RegisterClusterWithToken but supply extra ProviderInfo metadata.
+		Same functionality as RegisterClusterWithToken but supply extra ProviderInfo metadata and registration settings (namespace and policyRules).
```

Comment:
nit: stale comment? maybe:
```suggestion
		Same functionality as RegisterClusterWithToken but supply extra RegistrationMetadata.
```

Since we already have registration opts, do we also need registration metadata? Seems to me like we could nest registation meta within the Options struct and remove this provider-specific function altogether.

Reply:
Great suggestion, see the latest commit for the consolidated struct, a lot cleaner this way imo.